crates/edr_provider/tests/eip7702.rs

```diff
@@ -0,0 +1,361 @@
+#![cfg(feature = "test-utils")]
+
+#[path = "eip7702/different_sender_and_authorizer.rs"]
+mod different_sender_and_authorizer;
+#[path = "eip7702/invalid_chain_id.rs"]
+mod invalid_chain_id;
+#[path = "eip7702/invalid_nonce.rs"]
+mod invalid_nonce;
+#[path = "eip7702/multiple_authorizers.rs"]
+mod multiple_authorizers;
+#[path = "eip7702/reset.rs"]
+mod reset;
+#[path = "eip7702/same_sender_and_authorizer.rs"]
+mod same_sender_and_authorizer;
+#[path = "eip7702/zeroed_chain_id.rs"]
+mod zeroed_chain_id;
+
+use std::{convert::Infallible, sync::Arc};
+
+use edr_eth::{
+    eips::eip7702,
+    signature::{SecretKey, SignatureWithYParity},
+    Address, Bytes,
+};
+use edr_provider::{
+    time::CurrentTime, MethodInvocation, NoopLogger, Provider, ProviderConfig, ProviderRequest,
+};
+use edr_solidity::contract_decoder::ContractDecoder;
+use tokio::runtime;
+
+const CHAIN_ID: u64 = 0x7a69;
+
+fn assert_code_at(provider: &Provider<Infallible>, address: Address, expected: &Bytes) {
+    let code: Bytes = {
+        let response = provider
+            .handle_request(ProviderRequest::Single(MethodInvocation::GetCode(
+                address, None,
+            )))
+            .expect("eth_getCode should succeed");
+
+        serde_json::from_value(response.result).expect("response should be Bytes")
+    };
+
+    assert_eq!(code, *expected);
+}
+
+fn new_provider(config: ProviderConfig) -> anyhow::Result<Provider<Infallible>> {
+    let logger = Box::new(NoopLogger);
+    let subscriber = Box::new(|_event| {});
+
+    let provider = Provider::new(
+        runtime::Handle::current(),
+        logger,
+        subscriber,
+        config,
+        Arc::<ContractDecoder>::default(),
+        CurrentTime,
+    )?;
+
+    Ok(provider)
+}
+
+fn sign_authorization(
+    authorization: eip7702::Authorization,
+    secret_key: &SecretKey,
+) -> anyhow::Result<eip7702::SignedAuthorization> {
+    let signature = SignatureWithYParity::with_message(authorization.signature_hash(), secret_key)?;
+
+    Ok(authorization.into_signed(signature.into_inner()))
+}
+
+// #[tokio::test(flavor = "multi_thread")]
```

Comment:
Is this commented out on purpose?

Reply:
Resolved by https://github.com/NomicFoundation/edr/pull/805/commits/56a5820188862626d92a325e4c84fb1df869786c